### PR TITLE
update meta.yaml to 2.3.2 and modernize

### PR DESCRIPTION
**Pull request by mattip**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [x] Ensured the license file is being packaged.

- fix typo in license SPDX
- update to 2.3.2
- add minimum python version
- tweak recipe like in PR #9

### CODE REVIEW

**Comment by mattip** [comment]
@conda-forge-admin, please rerender

**Comment by mattip** [comment]
Hi @goanpeca I see you were one of the last people to merge a PR to this repo. Are you still a maintainer here? I noticed the package was not updating and put together a PR to clean it up a bit. The upstream package is maintained by @ianare, but I don't think they have maintainer powers here.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,20 +9,21 @@ package:
...
-  number: 1
-  script: python -m pip install --no-deps --ignore-installed .
+  number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
 
 requirements:
   host:
     - pip
     - python
+    - python >=3.4
```
Comment (mattip) [inline recipe/meta.yaml:23]: Conda wants a minimum version. Since conda does not support 2.7, I set it to 3.4+. Is that right?

**Comment by anjos** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (anjos) [reply recipe/meta.yaml:23]: I think you should remove this line.

Comment (anjos) [reply recipe/meta.yaml:23]: Actually, I'm wrong 😄!  You should actually constrain it on both the host and run sections.  So, you should remove line 22 and add something like line 23 (may be `python >=3` is more adequate?).  Then, you should also do that on the `run` section.  Instructions from here: https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python

**Pull request closed without merging**